File: frontend/src/api/labInfo.ts
import { api } from "./client"

export interface LabInfo {
  id: number
  company_name: string
  address: string
  city: string
  state: string
  zip_code: string
  phone: string
  email: string
  logo_url: string | null
  signature_url: string | null
  signer_name: string | null
  require_pdf_for_submission: boolean
  show_spec_preview_on_sample: boolean
  created_at: string
  updated_at: string
}

export interface LabInfoUpdate {
  company_name: string
  address: string
  phone?: string
  email?: string
  city: string
  state: string
  zip_code: string
  require_pdf_for_submission?: boolean
  show_spec_preview_on_sample?: boolean
}

export const labInfoApi = {
  /**
   * Get the current lab info configuration.
   */
  get: async (): Promise<LabInfo> => {
    const response = await api.get<LabInfo>("/settings/lab-info")
    return response.data
  },

  /**
   * Update lab info text fields.
   */
  update: async (data: LabInfoUpdate): Promise<LabInfo> => {
    const response = await api.put<LabInfo>("/settings/lab-info", data)
    return response.data
  },

  /**
   * Upload a new logo.
   */
  uploadLogo: async (file: File): Promise<LabInfo> => {
    const formData = new FormData()
    formData.append("file", file)
    const response = await api.post<LabInfo>("/settings/lab-info/logo", formData, {
      headers: {
        "Content-Type": "multipart/form-data",
      },
    })
    return response.data
  },

  /**
   * Delete the current logo.
   */
  deleteLogo: async (): Promise<LabInfo> => {
    const response = await api.delete<LabInfo>("/settings/lab-info/logo")
    return response.data
  },
}

File: backend/app/schemas/lab_info.py
"""Lab info schemas for request/response validation."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class LabInfoUpdate(BaseModel):
    """Schema for updating lab info."""

    company_name: str = Field(..., min_length=1, max_length=200)
    address: str = Field("", max_length=500)
    phone: Optional[str] = Field(None, max_length=50)
    email: Optional[str] = Field(None, max_length=200)
    city: str = Field("", max_length=100)
    state: str = Field("", max_length=50)
    zip_code: str = Field("", max_length=20)
    require_pdf_for_submission: Optional[bool] = None


class LabInfoResponse(BaseModel):
    """Lab info response schema."""

    id: int
    company_name: str
    address: str
    city: str
    state: str
    zip_code: str
    phone: str
    email: str
    logo_url: Optional[str] = None  # Computed full URL for frontend
    signature_url: Optional[str] = None  # Computed full URL for signature
    signer_name: Optional[str] = None
    require_pdf_for_submission: bool = True
    created_at: datetime
    updated_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class LogoUploadResponse(BaseModel):
    """Response after logo upload."""

    logo_url: str
    filename: str
